### PR TITLE
feat(dev-infra): make ts-api-guardian compatible with rules_nodejs version 3

### PR DESCRIPTION
**Pull request by alan-agius4**
Changes include:
- Remove `bazelWorkspaces` as this is no longer supported due to a single version policy https://github.com/bazelbuild/rules_nodejs/pull/783 and `install_bazel_dependencies` has been removed.
- Enable `bazel_patch_module_resolver` since monkey-patch the implementation of `require()` in NodeJS is opt-in

More info about the changes can be found here: https://github.com/bazelbuild/rules_nodejs/wiki#migrating-to-30

Blocking: https://github.com/angular/angular-cli/pull/19716

### CODE REVIEW

File: tools/ts-api-guardian/bin/ts-api-guardian [hunk 1]
```diff
@@ -1,3 +1,2 @@
 #!/usr/bin/env node
-
```
Comment (petebacondarwin) [inline tools/ts-api-guardian/bin/ts-api-guardian:2]: Unnecessary change?

Comment (alan-agius4) [reply tools/ts-api-guardian/bin/ts-api-guardian]: Reverted.

**Review by josephperrott — APPROVED** [review]
LGTM
